src/main/java/com/salesforce/dynamodbv2/mt/mappers/MtAmazonDynamoDbStreamsByTable.java

```diff
@@ -17,9 +23,31 @@
     }
 
     @Override
-    protected Function<Record, MtRecord> getRecordMapper(StreamArn arn) {
-        String[] tenantAndTableName = mtDynamoDb.getTenantAndTableName(arn.getTableName());
-        return record -> mapRecord(tenantAndTableName[0], tenantAndTableName[1], record);
+    protected MtGetRecordsResult getMtRecords(GetRecordsRequest request, StreamArn streamArn) {
+        final String[] tenantAndTableName = mtDynamoDb.getTenantAndTableName(streamArn.getTableName());
+        return getMtRecords(request, tenantAndTableName[0], tenantAndTableName[1]);
+    }
+
+    @Override
+    protected MtGetRecordsResult getMtRecords(GetRecordsRequest request, MtStreamArn mtStreamArn) {
+        return getMtRecords(request, mtStreamArn.getContext(), mtStreamArn.getTenantTableName());
+    }
+
+    private final MtGetRecordsResult getMtRecords(GetRecordsRequest request, String tenant, String tenantTable) {
+        final GetRecordsResult result = dynamoDbStreams.getRecords(request);
+        final List<Record> records = result.getRecords();
+        final String nextIterator = result.getNextShardIterator();
+        if (records.isEmpty()) {
+            return new MtGetRecordsResult().withRecords(records).withNextShardIterator(nextIterator);
+        }
+        final List<Record> mtRecords = new ArrayList<>(records.size());
+        for (Record record : records) {
+            mtRecords.add(mapRecord(tenant, tenantTable, record));
+        }
+        return new MtGetRecordsResult()
+            .withRecords(mtRecords)
+            .withNextShardIterator(nextIterator)
+            .withLastSequenceNumber(getLast(mtRecords).getDynamodb().getSequenceNumber());
     }
 
     private MtRecord mapRecord(String tenant, String tableName, Record record) {
```

Comment:
Mapping records for byTable is straightforward (and unchanged): only adds tenant id and table name to record (no field mapping needed).